src/Sentry.AspNetCore/SentryTunnelMiddleware.cs

```diff
@@ -98,10 +98,14 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
                     Method = HttpMethod.Post,
                     Content = new StreamContent(memoryStream),
                 };
+                var existingForwardedFor = context.Request.Headers["X-Forwarded-For"];
                 var clientIp = context.Connection?.RemoteIpAddress?.ToString();
                 if (clientIp != null)
                 {
-                    sentryRequest.Headers.Add("X-Forwarded-For", context.Connection?.RemoteIpAddress?.ToString());
+                    var forwardedFor = string.IsNullOrEmpty(existingForwardedFor)
```

Comment:
question: Is the `"X-Forwarded-For"` always either none or a single string, possibly comma-separated ... or could this represent multiple values, actually?

When the `StringValues` represents more than 1 string, then the implicit conversion in `string.IsNullOrEmpty` would allocate a new string (from the internal string-array).
In this case, perhaps `StringValues.Count` could be more efficient to detect whether there is any value existing.